### PR TITLE
Show notification when using shortcut to change current selected label to largest used label plus one

### LINKED ISSUE #6517: Keybinding to set a new label doesn't work consistently
### 🐛 Bug Report

After creating a labels layer, I observed that the `m` keybinding does not behave consistently.

### 💡 Steps to Reproduce

Create a labels layer.

- If the layer is selected, the `m` keybinding will either do nothing or select another layer wiht a name that starts with the letter `m`;
- If the paintbrush, fill bucket or polygon tools are selected, the `m` keybinding doesn't do anything;
- If the paintbrush has been used in the canvas, the `m` keybinding correctly adds 1 to the label control and a new label is selected and can be used to paint.

### 💡 Expected Behavior

The `m` keybinding should increase the label number in all those situations.

### 🌎 Environment

napari: 0.5.0a2.dev486+g4d60a7ce
Platform: Linux-6.1.64-1-MANJARO-x86_64-with-glibc2.38
System: Manjaro Linux
Python: 3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:40:35) [GCC 12.3.0]
Qt: 5.15.2
PyQt5: 5.15.10
NumPy: 1.26.2
SciPy: 1.11.4
Dask: 2023.11.0
VisPy: 0.14.1
magicgui: 0.8.0
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.2
npe2: 0.7.3

OpenGL:
  - GL version:  4.6 (Compatibility Profile) Mesa 23.1.9-manjaro1.1
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1920x1080, scale 1.0
  - screen 2: resolution 1920x1080, scale 1.0

Settings path:
  - /home/melissa/.config/napari/napari-dev_f5bfbd9c5d96bcb503f816d91f8db95d3b6d554f/settings.yaml
Plugins:
  - napari: 0.5.0a2.dev486+g4d60a7ce (77 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-svg: 0.1.10 (2 contributions)

### 💡 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dalthviz**
# References and relevant issues

Closes #6517 

# Description

Add showing a notification when the shortcut to set current selected label to largest used label plus one can't change the label value since it is already the largest used label plus one.

A preview:

label

### CODE REVIEW

File: napari/layers/labels/_labels_key_bindings.py [hunk 1]
```diff
@@ -74,7 +75,15 @@ def activate_labels_erase_mode(layer: Labels):
 )
 def new_label(layer: Labels):
     """Set the currently selected label to the largest used label plus one."""
-    layer.selected_label = np.max(layer.data) + 1
+    new_selected_label = np.max(layer.data) + 1
+    if layer.selected_label == new_selected_label:
+        show_info(
+            trans._(
+                "Largest label already selected",
```
Comment (Czaki) [inline napari/layers/labels/_labels_key_bindings.py]: It sounds misinformation to me. Largest may mean Largest by volume/area. 
And here we select first empty labels num.

Comment (dalthviz) [reply napari/layers/labels/_labels_key_bindings.py]: Maybe the message could be then something like `Maximum label value plus one already set as current selected label` or something like that which explains the situation with a more specific/correct wording? Not totally sure what words will describe better that `layer.selected_label` is already equal to `np.max(layer.data) + 1` 🤔

Comment (Czaki) [reply napari/layers/labels/_labels_key_bindings.py]: This proposition is much better.

Comment (psobolewskiPhD) [reply napari/layers/labels/_labels_key_bindings.py]: Yeah it's tricky.
`Maximum label value plus one already set as current selected label` isn't exactly intuitive either.
maybe `highest unused label`?

Comment (dalthviz) [reply napari/layers/labels/_labels_key_bindings.py]: So something like `Highest unused label already set as current selected label`?

Comment (Czaki) [reply napari/layers/labels/_labels_key_bindings.py]: No. Higest unused often is `np.iinfo(dtype).max`

maybe 

`Highest used label plus one already set as current selected label`

Comment (psobolewskiPhD) [reply napari/layers/labels/_labels_key_bindings.py]: I'd say that's an implementation detail? `highest used plus 1` will mean `highest unused` 99%of the time and the later is much clearer to someone not into the nitty gritty of how we implement labels

Comment (Czaki) [reply napari/layers/labels/_labels_key_bindings.py]: `higest used plus 1` will mean `lower unused` in 99% of the time. But if there is any hole in used labels, then it should land in this hole.

Comment (psobolewskiPhD) [reply napari/layers/labels/_labels_key_bindings.py]: Depends which way you look at it.
You're saying lowest of the labels that are unused, so from the PoV of the extant labels.
I'm looking from PoV of the user and the widget, where if you've used 1, 2, 3, then 4 is highest unused.

Comment (psobolewskiPhD) [reply napari/layers/labels/_labels_key_bindings.py]: In the end we need a message -- notification -- that is targeted at a user painting labels and is clear as to why seemingly nothing happened when they pushed the keybind.

Comment (dalthviz) [reply napari/layers/labels/_labels_key_bindings.py]: Maybe something like `Current selected label is not being used. You will need to use it to be able to set the current select label to the next label available` ? Kind of wordy but maybe can sound like more aimed to an end user which is unaware of implementation details but still wants to increase the current selected label? Could it even be worthy to mention in the notification that to increase the label regardless of it being used you can use the `=` shortcut?

Edit - The message with the `Increase the currently selected label by one` shortcut mention could be something like:

_Current selected label is not being used. You will need to first use it to be able to set the current select label to the next value available._

_To use the next label available regardles of it being used, please check the **Increase the currently selected label by one** keybinding (**=**)_

Comment (psobolewskiPhD) [reply napari/layers/labels/_labels_key_bindings.py]: I like that!

Comment (dalthviz) [reply napari/layers/labels/_labels_key_bindings.py]: What do you think about structuring the message [in that way](https://github.com/napari/napari/pull/6546#discussion_r1432857185) @Czaki ?

Comment (Czaki) [reply napari/layers/labels/_labels_key_bindings.py]: looks ok

**Comment by Czaki** [comment]
This PR will may generate conflict with #6548, where, because of typing, I add if guard to allow use only this action is a NumPy array (As it is a whole data scan). 

This is only a note, I do not thnk that it will require any action.

**Review by psobolewskiPhD — APPROVED** [review]

**Comment by psobolewskiPhD** [comment]
Just FYI, I think the plan is to merge stuff once 0.4.19 is released, to minimize cherry pick issues.

**Comment by psobolewskiPhD** [comment]
@Czaki we in the clear to merge this? Re: your comment above

**Review by Czaki — APPROVED** [review]

**Pull request merged**